### PR TITLE
feat(poems layout): hide sidebar on scroll

### PR DESCRIPTION
**Pull request by d4rekanguok**
Hey @andypbrowne! This PR hides the sidebar after scrolling x pixels, and shows it again when scrolling back up.

However when scrolling up, the sidebar appeared, but there was no way to reach its top (the category filter) without scrolling the page all the way up. This made it less useful (the sidebar has no context), so I made it sticky & put it in its own container. What do you think?

The design & transition might need some finessing, I'll leave that to you!

Here's a screen cap to save you a trip to netlify preview:

Screen Recording 2021-11-22 at 12 52 47 AM

### CODE REVIEW

**Comment by andypbrowne** [comment]
Is there an easy way for this feature to only apply on larger screens? I think the site is already doing something like that to control when the sidebar appears to the left or below the main content. 

I noticed that poems are hard to navigate on a small screen, but I also need to do more testin. 

Overall tho amazing! Hope we can get it to work well!

**Comment by d4rekanguok** [comment]
> Is there an easy way for this feature to only apply on larger screens? I think the site is already doing something like that to control when the sidebar appears to the left or below the main content.

Oops sorry about that! I've put in a fix to ensure we only track scroll on larger screen size. There were also a weird issues where clicking a link in the side bar _does not_ reset the window scroll position, so I put in a fix.. but need to investigate more https://github.com/sveltejs/kit/issues/2733

**Comment by andypbrowne** [comment]
Oh, by the way, making the sidebar sticky was a really good choice. I think its a nice way to navigate the list without the poem scrolling as well. And it looks clean too!

**Comment by d4rekanguok** [comment]
> Oh, by the way, making the sidebar sticky was a really good choice. I think its a nice way to navigate the list without the poem scrolling as well. And it looks clean too!

Yeah! I also noticed it's difficult at times to know if the list has ended, perhaps we could add a fade out or something to indicate there're more to the list

File: web/src/routes/poems/__layout.svelte [hunk 1]
```diff
@@ -40,62 +42,104 @@
...
+  let lastScrollY: number
+
+  let innerWidth = 500
+  let showSideBar = true
+  $: {
+    // if scrollY === lastScrollY, change nothing
+    showSideBar =
+      scrollY === lastScrollY
+        ? showSideBar
+        : scrollY < 500 || scrollY < lastScrollY
```
Comment (andypbrowne) [inline web/src/routes/poems/__layout.svelte:58]: Can we hide the sidebar faster by chaning just this one value. I'm guessing that `< 350` might get us close to the spot where the poem mainImage ends?

Comment (d4rekanguok) [reply web/src/routes/poems/__layout.svelte]: Yes, that is the number we can change to hide the side bar faster — I'll give 350 a try

**Comment by andypbrowne** [comment]
> > Oh, by the way, making the sidebar sticky was a really good choice. I think its a nice way to navigate the list without the poem scrolling as well. And it looks clean too!
> 
> Yeah! I also noticed it's difficult at times to know if the list has ended, perhaps we could add a fade out or something to indicate there're more to the list

Oh yes, I kinda like that idea of fading things. I'm picturing something like the social card I designed where the text kinda fades out at the bottom on the sidebar. Something like this picture. I'll admit… I don't know how to do that right away. 

Example - https://www.figma.com/file/KkfU6Scx2GhBWb4yG0QaJA/Karla-Kratz?node-id=827%3A6305

File: web/src/routes/poems/__layout.svelte (hunk 1 above)
Comment (andypbrowne) [reply web/src/routes/poems/__layout.svelte]: 350 actually works really well. the mainImage is just starting to scroll out of the viewport and then the sidebar goes too.

**Pull request merged**